### PR TITLE
Fix icon display

### LINKED ISSUE #32: Add-on icons in panel are broken
Because the panel does not have access to URLs like `jar:file:///var/folders/lq/xzbvpxjs4bj_m83qqkkn0bx40000gn/T/589bb220-5407-4242-80b2-f1b38fc481ca/extensions/compatibility@addons.mozilla.org.xpi!/icon.png`.

One idea would be to XHR the icon from `lib/main.js` and use the _base64_ string of it...

### PR DESCRIPTION
**Pull request by wagnerand**
Fixes #32.

Incredible how hard this was...

### CODE REVIEW

**Comment by wagnerand** [comment]
Since tofumatt is out, could you give this a look, please @kewisch? Thanks!

File: lib/main.js [hunk 1]
```diff
@@ -245,35 +249,61 @@ var reporterPanel = panel.Panel({
...
 
+function retrieveAddonIcon(addon, callback) {
+    if (!addon.iconURL) {
+        addon.icon = genericAddonIconURL;
+        callback(addon);
+    } else {
+        var xhr = new XMLHttpRequest();
+        xhr.onreadystatechange = function () {
+            if (this.readyState === 4) {
+            addon.icon = 'data:image/png;base64,' + base64.encode(String.fromCharCode.apply(null, new Uint8Array(this.response)));
```
Comment (kewisch) [inline lib/main.js]: You should test this for large images. I believe in some environments there is a limit as to how many arguments can be passed. Maybe use `(new UInt8Array(this.response)).map(c => String.fromCharCode(c)).join('')` or use `.subarray()` to split it into chunks.

Comment (wagnerand) [inline lib/main.js]: Icons are 32x32px, so that shouldn't become too big.

File: lib/main.js [hunk 2]
```diff
@@ -245,35 +249,61 @@ var reporterPanel = panel.Panel({
...
+                        addonReport.state = 3;
+                    if (addonReport.state == 0)
+                        addonReport.state = 1;
+                    addonReport.isDisabled = !addon.isActive;
+                    addonReport.multiprocessCompatible = addon.multiprocessCompatible;
+                    addonReports.push(addonReport);
+                    reportsProcessed++;
+
+                    if (reportsProcessed === addonsCount) {
+                        var by = addonReports.length*55;
```
Comment (kewisch) [inline lib/main.js]: `var by = Math.max(110, Math.min(220, addonReports.length * 55));` if you like.

**Comment by kewisch** [comment]
@wagnerand r+ with comments considered

File: lib/main.js (hunk 2 above)
Comment (wagnerand) [inline lib/main.js]: Oh, I just moved that whole pile of code. I am not touching that since the whole panel sizing is not trivial.

**Comment by wagnerand** [comment]
Thank you!

**Pull request merged**